// ang-jsoneditor/esm5/ang-jsoneditor.js
import { Component, Input, ViewChild, NgModule } from '@angular/core';
import * as editor from 'jsoneditor';
import { CommonModule } from '@angular/common';

var JsonEditorComponent = /** @class */ (function () {
    function JsonEditorComponent() {
        this._data = {};
        this.options = new JsonEditorOptions();
    }
    Object.defineProperty(JsonEditorComponent.prototype, "data", {
        set: function (value) {
            this._data = value;
            if (this.editor) {
                this.editor.destroy();
                this.ngOnInit();
            }
        },
        enumerable: true,
        configurable: true
    });
    JsonEditorComponent.prototype.ngOnInit = function () {
        this.editor = new editor(this.jsonEditorContainer.nativeElement, this.options, this._data);
    };
    JsonEditorComponent.prototype.collapseAll = function () {
        this.editor.collapseAll();
    };
    JsonEditorComponent.prototype.expandAll = function () {
        this.editor.expandAll();
    };
    JsonEditorComponent.prototype.focus = function () {
        this.editor.focus();
    };
    JsonEditorComponent.prototype.get = function () {
        return this.editor.get();
    };
    JsonEditorComponent.prototype.getMode = function () {
        return (this.editor.getMode());
    };
    JsonEditorComponent.prototype.getName = function () {
        return this.editor.getName();
    };
    JsonEditorComponent.prototype.getText = function () {
        return this.editor.getText();
    };
    JsonEditorComponent.prototype.set = function (json) {
        this.editor.set(json);
    };
    JsonEditorComponent.prototype.setMode = function (mode) {
        this.editor.setMode(mode);
    };
    JsonEditorComponent.prototype.setName = function (name) {
        this.editor.setName(name);
    };
    JsonEditorComponent.prototype.setSchema = function (schema) {
        this.editor.setSchema(schema);
    };
    JsonEditorComponent.prototype.setOptions = function (newOptions) {
        if (this.editor) {
            this.editor.destroy();
        }
        this.options = newOptions;
        this.ngOnInit();
    };
    JsonEditorComponent.prototype.destroy = function () {
        this.editor.destroy();
    };
    return JsonEditorComponent;
}());
JsonEditorComponent.decorators = [
    { type: Component, args: [{
                selector: 'json-editor',
                template: '<div #jsonEditorContainer></div>'
            },] },
];
JsonEditorComponent.ctorParameters = function () { return []; };
JsonEditorComponent.propDecorators = {
    "jsonEditorContainer": [{ type: ViewChild, args: ['jsonEditorContainer',] },],
    "options": [{ type: Input },],
    "data": [{ type: Input, args: ['data',] },],
};
var JsonEditorOptions = /** @class */ (function () {
    function JsonEditorOptions() {
        this.escapeUnicode = false;
        this.sortObjectKeys = false;
        this.history = true;
        this.mode = 'tree';
        this.search = true;
        this.indentation = 2;
    }
    return JsonEditorOptions;
}());
var NgJsonEditorModule = /** @class */ (function () {
    function NgJsonEditorModule() {
    }
    NgJsonEditorModule.forRoot = function () {
        return {
            ngModule: NgJsonEditorModule,
            providers: []
        };
    };
    return NgJsonEditorModule;
}());
NgJsonEditorModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    JsonEditorComponent
                ],
                exports: [
                    JsonEditorComponent
                ]
            },] },
];
NgJsonEditorModule.ctorParameters = function () { return []; };

export { NgJsonEditorModule, JsonEditorComponent, JsonEditorOptions };
